### PR TITLE
feat: add user schema

### PR DESCRIPTION
**Pull request by Ryszard-S**

### CODE REVIEW

File: packages/shared/models/user.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import mongoose from 'mongoose'
```
Comment (mcwiekala) [inline packages/shared/models/user.js:1]: please create test data

File: packages/shared/models/user.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+const userSchema = new mongoose.Schema({
+  id: String,
+  username: String,
+  passwordHash: String,
+  avatarURL: String,
```
Comment (mcwiekala) [inline packages/shared/models/user.js:7]: url? image content should be stored with type BSON 
https://docs.mongodb.com/manual/reference/bson-types/

File: packages/shared/models/user.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      id: String,
+      captionURL: String,
+      title: String
+    }
+  ],
+
+  users: [
+    {
+      id: String,
+      avatarURL: String
```
Comment (mcwiekala) [inline packages/shared/models/user.js:19]: BSON is too heavy for that...

File: packages/shared/models/user.js (hunk 2 above)
Comment (mcwiekala) [inline packages/shared/models/user.js:7]: change to avatarId: Number

File: packages/shared/models/user.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+const userSchema = new mongoose.Schema({
+  id: String,
+  username: String,
+  passwordHash: String,
+  avatarURL: String,
+  dashboards: [
+    {
+      id: String,
+      captionURL: String,
```
Comment (mcwiekala) [inline packages/shared/models/user.js:11]: change to imageCoverId: Number

File: packages/shared/models/user.js (hunk 3 above)
Comment (mcwiekala) [inline packages/shared/models/user.js:19]: avatarId: Number

File: packages/shared/models/user.js (hunk 4 above)
Comment (Ryszard-S) [reply packages/shared/models/user.js:11]: but component ImagePicker returns url to unsplash photo.

File: packages/shared/examples/user.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  passwordHash: 'pass123hash',
+  avatarId: 2,
+  dashboards: [
+    {
+      _id: ObjectId('622524b42dbfa0665c71fe9z'),
+      imageCoverId: 'photo-1640622842924-fb0017b9d786',
+      title: 'New board 1'
+    }
+  ],
+  users: [
```
Comment (mcwiekala) [inline packages/shared/examples/user.js:43]: user has a users? What is the reason for that?

File: packages/shared/models/user.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+const userSchema = new mongoose.Schema({
+  _id: ObjectId,
+  username: String,
+  passwordHash: String,
+  avatarId: Number,
+  dashboards: [
+    {
+      _id: ObjectId,
+      imageCoverId: String,
+      title: String
```
Comment (mcwiekala) [inline packages/shared/models/user.js:14]: Add role enum:
    type: String,
    enum: ['admin', 'member', 'viever'],
    
    
see: https://mongoosejs.com/docs/validation.html

File: packages/shared/models/user.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    }
+  ],
+
+  users: [
```
Comment (mcwiekala) [inline packages/shared/models/user.js:18]: user has a users? What is the reason for that?

**Pull request merged**